drivers/i2s/i2s_nrfx.h

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief Driver for Nordic Semiconductor nRF I2S
+ */
+
+
+
+#ifndef DRIVERS_I2S_I2S_NRFX_H_
+#define DRIVERS_I2S_I2S_NRFX_H_
+
+/** @brief I2S driver clock configuration structure. */
+struct i2s_clk_settings_t {
+	u32_t frequency;		/**< Configured frequency [Hz]. */
+	nrf_i2s_ratio_t ratio[3];	/**< Content of CONFIG.RATIO register
+					  * for given frequency. Every element
+					  * of ratio[3] array corresponds to
+					  * one of 3 possible sample width
+					  * values
+					  * (accordingly 8, 16 and 24 bit).
+					  */
+	nrf_i2s_mck_t divider[3];	/**< Content of CONFIG.MCKFREQ register
+					  * for given frequency. Every element
+					  * of divider[3] array corresponds to
+					  * one of 3 possible sample width
+					  * values
+					  *(accordingly 8, 16 and 24 bit).
+					  */
+};
+
+/**@brief I2S driver clock configfuration table. */
+
+#define NRFX_I2S_AVAILABLE_CLOCK_SETTINGS		\
```

Comment:
Perhaps it would be valuable to mention that the below settings do not provide exact values of the sampling frequencies because it is not possible with the hardware. Also, it could be good to mention why separate settings have to be provided for different sample widths (while in some cases they are exactly the same). In case someone would need to extend these settings, this would be helpful.